src/sdk/namespace_fs.js

```diff
@@ -1405,10 +1484,57 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * 
+     * @param {*} fs_context - fs context using to check symbolic links
+     * @param {*} file_path - path to file
+     * @param {*} set - the xattr object to be set
+     * @param {*} clear - the xattr prefix to be cleared
+     * @returns 
+     */
+    async set_fs_xattr_op(fs_context, file_path, set, clear) {
+        let file;
+        try {
+            file = await nb_native().fs.open(fs_context, file_path, undefined, get_umasked_mode(config.BASE_MODE_FILE));
+            await file.replacexattr(fs_context, set, clear);
+            await file.close(fs_context);
+            file = null;
+        } catch (error) {
+            dbg.error('namespace_fs.handle_fs_xattr_op: failed with error: ', error, file_path);
+            throw this._translate_object_error_codes(error);
+        } finally {
+            if (file) await file.close(fs_context);
+        }
+    }
+
+    // when deleting the data of a directory object, we need to remove the directory dir object xattr 
+    // if the dir still exists - occurs when deleting dir while the dir still has entries in it
+    async _clear_user_xattr(fs_context, params) {
+        const file_md_path = this._get_file_md_path(params);
+        try {
+            await this.set_fs_xattr_op(fs_context, file_md_path, undefined, XATTR_USER_PREFIX);
+        } catch (err) {
+            if (err.code !== 'ENOENT') throw err;
+            dbg.log0(`namespace_fs._clear_user_xattr: dir ${file_md_path} was already deleted`);
+        }
+    }
+
+    async _assign_xattr_to_dir_content(fs_context, fs_xattr, params) {
```

Comment:
add comments to all of the functions like this, as we won't remember what the name means

Reply:
done